Jellyfin.Api/Controllers/NotificationsController.cs

```diff
@@ -0,0 +1,176 @@
+    /// The notification controller.
+    /// </summary>
+    public class NotificationsController : BaseJellyfinApiController
+    {
+        private readonly INotificationManager _notificationManager;
+        private readonly IUserManager _userManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NotificationsController" /> class.
+        /// </summary>
+        /// <param name="notificationManager">The notification manager.</param>
+        /// <param name="userManager">The user manager.</param>
+        public NotificationsController(INotificationManager notificationManager, IUserManager userManager)
+        {
+            _notificationManager = notificationManager;
+            _userManager = userManager;
+        }
+
+        /// <summary>
+        /// Endpoint for getting a user's notifications.
+        /// </summary>
+        /// <param name="userId">The user's ID.</param>
+        /// <param name="isRead">An optional filter by IsRead.</param>
+        /// <param name="startIndex">The optional index to start at. All notifications with a lower index will be dropped from the results.</param>
+        /// <param name="limit">An optional limit on the number of notifications returned.</param>
+        /// <returns>A read-only list of all of the user's notifications.</returns>
+        [HttpGet("{UserID}")]
+        [ProducesResponseType(typeof(IEnumerable<NotificationResultDto>), StatusCodes.Status200OK)]
+        public IActionResult GetNotifications(
+            [FromRoute] string userId,
+            [FromQuery] bool? isRead,
+            [FromQuery] int? startIndex,
+            [FromQuery] int? limit)
+        {
+            return Ok(new NotificationResultDto());
+        }
+
+        /// <summary>
+        /// Endpoint for getting a user's notification summary.
+        /// </summary>
+        /// <param name="userId">The user's ID.</param>
+        /// <returns>Notifications summary for the user.</returns>
+        [HttpGet("{UserID}/Summary")]
+        [ProducesResponseType(typeof(NotificationsSummaryDto), StatusCodes.Status200OK)]
+        public IActionResult GetNotificationsSummary(
+            [FromRoute] string userId)
+        {
+            return Ok(new NotificationsSummaryDto());
+        }
+
+        /// <summary>
+        /// Endpoint for getting notification types.
+        /// </summary>
+        /// <returns>All notification types.</returns>
+        [HttpGet("Types")]
+        [ProducesResponseType(typeof(IEnumerable<NameIdPair>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status500InternalServerError)]
+        public IActionResult GetNotificationTypes()
+        {
+            try
+            {
+                return Ok(_notificationManager.GetNotificationTypes());
+            }
+            catch (Exception e)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, e.Message);
```

Comment:
We shouldn't handle exceptions manually like this in every single endpoint, there should be a global exception handler defined that does this. There already is one that exists whose behavior should be matched for compatability: https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/HttpServer/HttpListenerHost.cs#L242.

Reply:
Should creating that be a separate PR? If so, should I keep this as-is for now and modify it in that PR? If not, I'll just add it now.

Reply:
Yes, I think it would be best to remove the error handling here and add the global handler in a separate PR.